Extension/browser/safari/lib/content-blocker.js

```diff
@@ -79,6 +79,15 @@ var SafariContentBlocker = exports.SafariContentBlocker = {
     },
 
     _setContentBlocker: function (json) {
+        if (this.wait) {
+            this.json = json;
+            Log.info('Waiting to set content blocker');
+            return;
+        }
+
+        this.wait = true;
+        delete this.json;
+
         try {
```

Comment:
Delay execution of the block below, otherwise this "wait" logic simply does nothing